cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/namespace/DefaultNamespaceAdmin.java

```diff
@@ -236,6 +235,8 @@ public Void call() throws Exception {
     } catch (Exception e) {
       LOG.warn("Error while deleting namespace {}", namespaceId, e);
       throw new NamespaceCannotBeDeletedException(namespaceId, e);
+    } finally {
+      authorizerInstantiator.get().revoke(namespace);
```

Comment:
We should have a JIRA to do this transactionally. Currently if there are failures, we will revoke access but there can be partial information about the namespace.